api-client/src/main/java/com/takipi/api/client/request/advanced/UpdateAdvancedSettingsRequest.java

```diff
@@ -0,0 +1,72 @@
+package com.takipi.api.client.request.advanced;
+
+import java.io.UnsupportedEncodingException;
+
+import com.google.gson.Gson;
+import com.takipi.api.client.data.advanced.AdvancedSettings;
+import com.takipi.api.client.request.ServiceRequest;
+import com.takipi.api.client.result.EmptyResult;
+import com.takipi.api.core.consts.ApiConstants;
+import com.takipi.api.core.request.intf.ApiPostRequest;
+
+public class UpdateAdvancedSettingsRequest extends ServiceRequest implements ApiPostRequest<EmptyResult> {
+	private final AdvancedSettings advancedSettings;
+	
+	protected UpdateAdvancedSettingsRequest(String serviceId, AdvancedSettings advancedSettings)
+	{
+		super(serviceId);
+		
+		this.advancedSettings = advancedSettings;
+	}
+	
+	@Override
+	public byte[] postData() throws UnsupportedEncodingException {
+		return (new Gson()).toJson(this.advancedSettings).getBytes(ApiConstants.UTF8_ENCODING);
+	}
+	
+	@Override
+	public Class<EmptyResult> resultClass() {
+		return EmptyResult.class;
+	}
+	
+	@Override
+	public String urlPath() {
+		return baseUrlPath() + "/settings/advanced-settings";
+	}
+	
+	public static Builder newBuilder() {
+		return new Builder();
+	}
+	
+	public static class Builder extends ServiceRequest.Builder {
+		private AdvancedSettings advancedSettings;

```

Comment:
This field should be replaced by all the internal fields of `AdvancedSettings`

We want to allow an easy creation of the request with just the parameters the user wants to set.